tests/components/nest/test_events.py

```diff
@@ -0,0 +1,237 @@
+            "eventId": "some-event-id",
+            "timestamp": utcnow().isoformat(timespec="seconds"),
+            "resourceUpdate": {
+                "name": device_id,
+                "events": events,
+            },
+        },
+        auth=None,
+    )
+
+
+async def test_doorbell_chime_event(hass):
+    """Test a pubsub message for a doorbell event."""
+    events = async_capture_events(hass, NEST_EVENT)
+    subscriber = await async_setup_devices(
+        hass,
+        "sdm.devices.types.DOORBELL",
+        create_device_traits("sdm.devices.traits.DoorbellChime"),
+    )
+
+    registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = registry.async_get("camera.front")
+    assert entry is not None
+    assert entry.unique_id == "some-device-id-camera"
+    assert entry.original_name == "Front"
+    assert entry.domain == "camera"
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+    assert device.name == "Front"
+    assert device.model == "Doorbell"
+    assert device.identifiers == {("nest", DEVICE_ID)}
+
+    await subscriber.async_receive_event(
+        create_event("sdm.devices.events.DoorbellChime.Chime")
+    )
+    await hass.async_block_till_done()
+
+    assert len(events) == 1
+    assert {
+        "device_id": entry.device_id,
+        "type": "DoorbellChime",
+    } == events[0].data
+
+
+async def test_camera_motion_event(hass):
+    """Test a pubsub message for a camera motion event."""
+    events = async_capture_events(hass, NEST_EVENT)
+    subscriber = await async_setup_devices(
+        hass,
+        "sdm.devices.types.CAMERA",
+        create_device_traits("sdm.devices.traits.CameraMotion"),
+    )
+    registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = registry.async_get("camera.front")
+    assert entry is not None
+
+    await subscriber.async_receive_event(
+        create_event("sdm.devices.events.CameraMotion.Motion")
+    )
+    await hass.async_block_till_done()
+
+    assert len(events) == 1
+    assert {
+        "device_id": entry.device_id,
+        "type": "CameraMotion",
+    } == events[0].data
+
+
+async def test_camera_sound_event(hass):
+    """Test a pubsub message for a camera sound event."""
+    events = async_capture_events(hass, NEST_EVENT)
+    subscriber = await async_setup_devices(
+        hass,
+        "sdm.devices.types.CAMERA",
+        create_device_traits("sdm.devices.traits.CameraSound"),
+    )
+    registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = registry.async_get("camera.front")
+    assert entry is not None
+
+    await subscriber.async_receive_event(
+        create_event("sdm.devices.events.CameraSound.Sound")
+    )
+    await hass.async_block_till_done()
+
+    assert len(events) == 1
+    assert {
+        "device_id": entry.device_id,
+        "type": "CameraSound",
+    } == events[0].data
```

Comment:
This is really minor but the expected value should be on the right of `==`
```suggestion
    assert events[0].data == {
        "device_id": entry.device_id,
        "type": "CameraSound",
    }
```